crates/transaction-generator-lib/src/publishing/prebuild_packages.rs

```diff
@@ -23,255 +26,133 @@ fn get_local_framework_path() -> String {
-
-    let mut generic_mod = std::fs::File::create(output_file).unwrap();
-    {
-        use std::io::Write;
-        write!(generic_mod, "{}", string_buffer)?;
+        build_options
     }
-    Ok(())
 }
 
-// Write out given package
-fn write_package(
-    file: &mut String,
-    package_path: PathBuf,
-    package_name: &str,
-    build_options: BuildOptions,
-) -> (String, bool) {
-    println!("Building package {}", package_name);
-    // build package
-    let package =
-        BuiltPackage::build(package_path, build_options).expect("building package must succeed");
-    let modules = package.extract_code();
-    let mut scripts = package.extract_script_code();
-    let package_metadata = package.extract_metadata().expect("Metadata must exist");
-    let metadata = bcs::to_bytes(&package_metadata).expect("Metadata must serialize");
-
-    // write out package metadata
-    write_lazy(
-        file,
-        format!("PACKAGE_{}_METADATA", package_name.to_uppercase()).as_str(),
-        &metadata,
-    );
-
-    let mut module_names = Vec::new();
-
-    // write out all modules
-    for module in &modules {
-        // this is an unfortunate way to find the module name but it is not
-        // clear how to do it otherwise
-        let compiled_module = CompiledModule::deserialize(module).expect("Module must deserialize");
-        let module_name = compiled_module.self_id().name().to_owned().into_string();
-        // start Lazy declaration
-        let name: String = format!(
-            "MODULE_{}_{}",
-            package_name.to_uppercase(),
-            module_name.to_uppercase()
-        );
-        writeln!(file).expect("Empty line failed");
-        write_lazy(file, name.as_str(), module);
-        module_names.push(name);
-    }
-
-    assert!(
-        scripts.len() <= 1,
-        "Only single script can be added per package"
-    );
-
-    let has_script = if let Some(script) = scripts.pop() {
-        let name: String = format!("SCRIPT_{}", package_name.to_uppercase());
-        writeln!(file).expect("Empty line failed");
-        write_lazy(file, name.as_str(), &script);
-        true
-    } else {
-        false
-    };
-
-    writeln!(file).expect("Empty line failed");
-    writeln!(file, "#[rustfmt::skip]").expect("rustfmt skip failed");
-    writeln!(
-        file,
-        "pub static MODULES_{}: Lazy<Vec<Vec<u8>>> = Lazy::new(|| {{ vec![",
-        package_name.to_uppercase(),
-    )
-    .expect("Lazy MODULES declaration failed");
+/// Creates a [PrebuiltPackagesBundle] from the provided list of packages, serializes it and saves
+/// as a file in the specified base directory (should be a Cargo crate). Also generates a Rust file
+/// in that crate that allows to access prebuilt information. The output file must live in the same
+/// crate.
+pub fn create_prebuilt_packages_bundle(
+    base_dir: impl AsRef<Path>,
+    packages_to_build: Vec<(PathBuf, PrebuiltPackageConfig)>,
+    output_file: impl AsRef<Path>,
```

Comment:
rename to output_rust_file for clarity.

and mention that output bundle is in base_dir/prebuilt.mpb in the commend